napari/components/add_layers_mixin.py

```diff
@@ -896,3 +900,98 @@ def _add_layer_from_data(
                 raise exc
 
         return layer
+
+    def _save_layers(
```

Comment:
while `save_selected` belongs on the `Viewer` object, I'm not sure this should live on the viewer object.  It doesn't need access to the viewer, just a list of layers.  I see this more as a top level `napari.save_layers` convenience function.  And, for that matter, i think `napari.plugins.io.write_multiple_layers_with_plugin` already IS that convenience function... so do we really need this one at all?

Reply:
the "if len(layers) > 1" logic does belong somewhere, but that should then be the main function we use:

```python
def save_layers(path, layers, plugin_name=None):
    if isinstance(layers, list):
        if len(layers) > 1:
            return write_multiple_layers(path, layers, plugin_name)
        elif len(layers) == 1:
            layers = layers[0]
    if isinstance(layers, Layer):
        return write_single_layer(path, layers, plugin_name)
```

Reply:
changes made, i think this moves in the right direction, let me know if you have more suggestions